# While Loop

In [1]:
print("Himanshu")

Himanshu


In [2]:
print("Himanshu")
print("Himanshu")
print("Himanshu")

Himanshu
Himanshu
Himanshu


In [3]:
i = 1

while i <= 10 :
    print("Himanshu")
    i += 1             # i = i + 1,   1, 2, 3 ....10, 11

Himanshu
Himanshu
Himanshu
Himanshu
Himanshu
Himanshu
Himanshu
Himanshu
Himanshu
Himanshu


In [4]:
i = 1

while i <= 10:
    print(i)
    i += 1

1
2
3
4
5
6
7
8
9
10


In [5]:
print(i)

11


In [6]:
print("Hello\nIndore")

Hello
Indore


In [7]:
print("Hello")
print("Indore")

Hello
Indore


In [8]:
print("Hello", end='\n')
print("World")

Hello
World


In [9]:
print("Hello", end='\t')
print("World")

Hello	World


In [10]:
print("Hello", end='+')
print("World")

Hello+World


In [11]:
print("Hello", end=' ')
print("World")

Hello World


In [12]:
print("Hello", end='')
print("World")

HelloWorld


In [ ]:
1  2  3  4  5  6  7  8  9  10

In [13]:
i = 1

while i <= 10:
    print(i, end='  ')
    i += 1

1  2  3  4  5  6  7  8  9  10  

## WAP to print following series

In [ ]:
1  4  9  16  25  36  49  64  81  100

In [ ]:
i ** 2

i * i

In [14]:
a = 10

print(a * 4)

print(a)

40
10


In [15]:
i = 1

while i <= 10:
    print(i * i, end='  ')
    i += 1

1  4  9  16  25  36  49  64  81  100  

In [ ]:
1 + 2 + 3 + 4 + 5 + 6 + 7 + 8 + 9 + 10

In [16]:
i = 1

while i <= 10:
    print(i, end=' + ')
    i += 1

1 + 2 + 3 + 4 + 5 + 6 + 7 + 8 + 9 + 10 + 

In [17]:
i = 1

while i <= 10:
    if i < 10:
        print(i, end=' + ')
    else:
        print(i)
    i += 1

1 + 2 + 3 + 4 + 5 + 6 + 7 + 8 + 9 + 10


In [ ]:
1/1  2/4  3/9  4/16  5/25  6/36 ........ 10/100

In [18]:
i = 1

while i <= 10:
    print(i,"/", i * i, end='  ')
    i += 1

1 / 1  2 / 4  3 / 9  4 / 16  5 / 25  6 / 36  7 / 49  8 / 64  9 / 81  10 / 100  

In [19]:
i = 1

while i <= 10:
    print(i, end='/')
    print(i * i, end='  ')
    i += 1

1/1  2/4  3/9  4/16  5/25  6/36  7/49  8/64  9/81  10/100  

In [22]:
# dani
i = 1   
while i <= 10:
    print(i, end="/")
    print(i * i, end='  ')
    i += 1

1/1  2/4  3/9  4/16  5/25  6/36  7/49  8/64  9/81  10/100  

## break statement

In [23]:
i = 1

while i <= 10:
    if i == 5:
        print("Hi")
        break
    print(i)
    i += 1

1
2
3
4
Hi


## continue statement

In [24]:
i = 0

while i <= 10:
    i += 1
    if i%2 == 0:
        continue
    else:
        print(i)

1
3
5
7
9
11


# Questions

**WAP to enter 5 values from user and print them.**

In [25]:
i = 1

while i <= 5:
    a = int(input("Enter Number: "))
    print("Your Number is :", a)
    
    i += 1

Enter Number: 10
Your Number is : 10
Enter Number: 20
Your Number is : 20
Enter Number: 30
Your Number is : 30
Enter Number: 40
Your Number is : 40
Enter Number: 50
Your Number is : 50


**WAP to enter 5 numbers from user and print sum of them**

In [ ]:
s = 0
s = s + a

In [27]:
i = 1
s = 0

while i <= 5:
    a = int(input("Enter Number: "))
    s = s + a
    i += 1

print("Summation =", s)

Enter Number: 10
Enter Number: 20
Enter Number: 30
Enter Number: 40
Enter Number: 50
Summation = 150


# 5 While loop

**WAP to enter a number from user and print reverse of it.**

153

351

In [ ]:
153 % 10  = 3
153 // 10 = 15

15 % 10  = 5
15 // 10 = 1

1 % 10 = 1
1 // 10 = 0

In [2]:
n = int(input("Enter a Number: "))

while n > 0:              # n = 153, 15, 1
    r = n % 10            # r = 3, 5, 1
    print(r, end='')      # 351
    n = n // 10           # n = 15, 1, 0

Enter a Number: 12345
54321

**WAP to enter a number from user and print sum of its individual digits**

153

1+5+3 = 9


In [ ]:
0
3 = s
5 + s = s
1 + s = s

In [5]:
n = int(input("Enter a Number: "))
s = 0

while n > 0:              # n = 153, 15, 1
    r = n % 10            # r = 3, 5, 1
    s = s + r             # s = 0, 3, 8, 9
    n = n // 10           # n = 15, 1, 0

print("Sum =", s)

Enter a Number: 121
Sum = 4


### reverse of the number

In [6]:
n = int(input("Enter a Number: "))
s = 0

while n > 0:              # n = 153, 15, 1
    r = n % 10            # r = 3, 5, 1
    s = s * 10 + r        # s = 0, 3, 35, 351
    n = n // 10           # n = 15, 1, 0

print("Reverse =", s)
print(s * 2)

Enter a Number: 153
Reverse = 351
702


**WAP to enter  a number form user and check if if is palindrome or not**

153 <=> 351 - not a palindrome

121 <=> 121 - palindrome

In [11]:
n = int(input("Enter a Number: "))
t = n
s = 0

while n > 0:              # n = 153, 15, 1
    r = n % 10            # r = 3, 5, 1
    s = s * 10 + r        # s = 0, 3, 35, 351
    n = n // 10           # n = 15, 1, 0

if s == t:
    print("Palindrome")
else:
    print("Not a Palindrome")

Enter a Number: 12321
Palindrome


**WAP to enter a number from user and check if it is armstrong number or not**

153 -> cube(1) + cube(5) + cube(3) -> 153

370 -> 

371

In [14]:
n = int(input("Enter a Number: "))
t = n
s = 0

while n > 0:              # n = 153, 15, 1
    r = n % 10            # r = 3, 5, 1
    s = s + r*r*r         # s = 0, 27, 152, 153
    n = n // 10           # n = 15, 1, 0

if s == t:
    print("Armstrong")
else:
    print("Not an Armstrong")

Enter a Number: 370
Armstrong


**WAP to enter a number from user and print its individual digits on seperate line**

153

1

5

3

In [15]:
n = int(input("Enter a Number: "))

while n > 0:              # n = 153, 15, 1
    r = n % 10            # r = 3, 5, 1
    print(r) 
    n = n // 10           # n = 15, 1, 0

Enter a Number: 153
3
5
1


In [16]:
n = int(input("Enter a Number: "))
s = 0

while n > 0:              # n = 153, 15, 1
    r = n % 10            # r = 3, 5, 1
    s = s * 10 + r 
    n = n // 10           # n = 15, 1, 0
    
while s > 0:
    r = s % 10
    print(r)
    s = s // 10

Enter a Number: 153
1
5
3


**WAP to enter a number from user and print individual digits in character format**

153

one

five

three

In [18]:
n = int(input("Enter a Number: "))
s = 0

while n > 0:              # n = 153, 15, 1
    r = n % 10            # r = 3, 5, 1
    s = s * 10 + r 
    n = n // 10           # n = 15, 1, 0
    
while s > 0:
    r = s % 10
    if r == 1:
        print("One")
    elif r == 2:
        print("Two")
    elif r == 3:
        print("Three")
    elif r == 4:
        print("Four")
    elif r == 5:
        print("Five")
    s = s // 10

Enter a Number: 123
One
Two
Three


**WAP to print factorial of a number**

!3 -> 3 * 2 * 1 = 6

!5 -> 5 * 4 * 3 * 2 * 1 = 120